### ISSUE TITLE
feat: Support `inline-policy` in iam-assumable-role

### ISSUE THREAD

**Issue by jmctune** [terraform-aws-modules/terraform-aws-iam#419]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality

I don't know which provider version initially introduced the functionality, but specifying an `inline_policy` has been core to IAM for years.

## Is your request related to a problem? Please describe.

Yes, it is currently not possible for me to use this module by itself and requires an accompanying resource to use. In my use case, I use terragrunt and would like to specify *just* the `iam-assumable-role` module as it has everything I need, except for the ability to specify a custom, inline policy.

## Describe the solution you'd like.

I'd like to see [inline_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#inline_policy) supported when calling `aws_iam_role` [here](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role/main.tf#L142), while allowing the user to specify a custom inline policy the same way they can feed in a custom [role trust policy](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role/main.tf#L9).

This would allow me to easily call this module in terragrunt without having to specify my own custom module to call this module with one additional managed policy like so:

```hcl
terraform {
  source = "git@github.com:terraform-aws-modules/terraform-aws-iam//modules/iam-assumable-role?ref=856b5438e91dd87c1e9d225afd123769aaadb635" # 5.30.0
}

inputs = {
  create_role = true
  role_name = "my_role"
  inline_policy = <my_json_encoded_policy>
}
```

This was originally requested in #261, but was closed due to inactivity.

## Describe alternatives you've considered.

The alternative is to create a secondary module that calls `iam-assumable-role` and a singular policy, then passes the created policy back to the module. I don't have a need to create a managed policy - just a single inline policy on the role.

## Additional context

N/A

**Comment by jmctune** [comment]
Bump.

**Comment by julb** [comment]
I confirm, this would be very useful.
Thanks for the awesome work.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 5.40.0 :tada:
